css/src/components/chip.css

```diff
@@ -137,12 +137,37 @@
   border-radius: 10px;
 }
 
+.Chip-icon--right:hover {
+  background-color: var(--secondary);
+  border-radius: 10px;
+}
+
+.Chip-icon--right:active {
+  background-color: var(--secondary-dark);
+  border-radius: 10px;
+}
+
+.Chip-icon-disabled--right {
+  pointer-events: none;
+}
+
 .Chip-icon--selected:focus-visible {
   outline: var(--spacing-s) solid var(--primary-shadow);
   border-radius: 10px;
 }
 
+.Chip-icon--selected:hover {
+  background-color: var(--primary-lighter);
+  border-radius: 10px;
+}
+
+.Chip-icon--selected:active {
+  background-color: var(--primary-light);
+  border-radius: 10px;
+}
+
```

Comment:
Please give `border-radius: 10px` css to `Chip-icon--selected` class directly instead of using the same css in different condition to remove redundancy